### PR TITLE
Revise PriorBoxClustered Op

### PR DESCRIPTION
**Pull request by blesniewski**
### Details:
 - *Separated type prop tests to a dedicated file, added few cases*
 - *Added visitor tests*
 - *`Step` was added as an attribute (there was a mismatch visible after adding attribute count check to visitor tests)*
 - *Added SLT for CPU, modified already present GPU SLT and SSLT to work with newly added `step` attribute*
 - *Added backend tests*
 - *Fixed the `variance` attribute problem- made sure to first check for the number of attributes, then copy the variance values an adequate number of times*
 - *Added `PriorBoxClustered` to trusted ops list*

### Tickets:
 - *37463, 48591*
 - *45983, 45984, 45985, 53988*

### CODE REVIEW

File: ngraph/core/include/ngraph/op/prior_box_clustered.hpp [hunk 1]
```diff
@@ -24,6 +24,7 @@ namespace ngraph
             bool clip = true;
             float step_widths = 0.0f;
             float step_heights = 0.0f;
+            float step = 0.0f;
```
Comment (pelszkow) [inline ngraph/core/include/ngraph/op/prior_box_clustered.hpp:27]: Missing `step` description.

**Review by pelszkow — APPROVED** [review]

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/prior_box_clustered.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+const std::vector<float> step_widths = {
+    0.0f, 2.0f
+};
+
+const std::vector<float> step_heights = {
+    0.0f, 1.5f
+};
+
+const std::vector<float> step = {
+    0.0f
```
Comment (ilyachur) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/prior_box_clustered.cpp:38]: Maybe do we need to check more values?

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/prior_box_clustered.cpp [hunk 3]
```diff
@@ -34,6 +34,10 @@ const std::vector<float> step_heights = {
     0.0f, 1.5f
 };
 
+const std::vector<float> step = {
+    0.0f
```
Comment (ilyachur) [inline inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/prior_box_clustered.cpp:38]: The same comment here

File: ngraph/core/include/ngraph/op/prior_box_clustered.hpp (hunk 1 above)
Comment (ilyachur) [reply ngraph/core/include/ngraph/op/prior_box_clustered.hpp:27]: We don't have the `step` attribute in the specification. Should we update the spec?

File: ngraph/core/reference/include/ngraph/runtime/reference/prior_box_clustered.hpp [hunk 4]
```diff
@@ -38,16 +39,11 @@ namespace ngraph
                 int64_t img_height = img[0];
 
                 // TODO: Uncomment after PriorBoxClustered is aligned with the specification.
```
Comment (ilyachur) [inline ngraph/core/reference/include/ngraph/runtime/reference/prior_box_clustered.hpp:41]: I think the operation revising should allow to align spec and op and remove this commented code.

File: ngraph/core/include/ngraph/op/prior_box_clustered.hpp (hunk 1 above)
Comment (itikhono) [reply ngraph/core/include/ngraph/op/prior_box_clustered.hpp:27]: I guess we have 3 similar attrubutes:` step_h, step_w and step`
image

Comment (ilyachur) [reply ngraph/core/include/ngraph/op/prior_box_clustered.hpp:27]: Yes my mistake.

Comment (blesniewski) [reply ngraph/core/include/ngraph/op/prior_box_clustered.hpp:28]: Added the description 👍

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/prior_box_clustered.cpp (hunk 2 above)
Comment (blesniewski) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/prior_box_clustered.cpp]: Added more values 👍

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/prior_box_clustered.cpp (hunk 3 above)
Comment (blesniewski) [reply inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/prior_box_clustered.cpp]: Added more values 👍

File: ngraph/core/reference/include/ngraph/runtime/reference/prior_box_clustered.hpp (hunk 4 above)
Comment (blesniewski) [reply ngraph/core/reference/include/ngraph/runtime/reference/prior_box_clustered.hpp]: Removed the obsolete comment 👍

File: ngraph/core/reference/include/ngraph/runtime/reference/prior_box_clustered.hpp [hunk 5]
```diff
@@ -37,17 +38,8 @@ namespace ngraph
                 int64_t img_width = img[1];
                 int64_t img_height = img[0];
 
-                // TODO: Uncomment after PriorBoxClustered is aligned with the specification.
```
[the remark this answers is not in the document]
Comment (blesniewski) [reply ngraph/core/reference/include/ngraph/runtime/reference/prior_box_clustered.hpp:40]: There are no such attributes anywhere in the code for PriorBoxClustered, single layer tests are passing for CPU locally for me, and also for GPU on the CI. I'm only removing an obsolete comment.

**Review by itikhono — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**